libraries/ms-common/src/fifo.c

```diff
@@ -0,0 +1,130 @@
+#include "fifo.h"
+#include <stdio.h>
```

Comment:
`<string.h>` should technically be here to include `memcpy` and `memset` and you don't appear to use anything from `<stdio.h>` so far as I can tell.